### ISSUE TITLE
Questions about the dataset 

### ISSUE THREAD

**Issue by yyuzhongpv** [ArthurFirmino/progressive-denoising#2]
Hi Arthur,

Thanks for your great work and sharing the dataset.

I have some quick questions about the dataset.

1. How did you generate the training dataset? You mentioned that you were using mitsuba to generate it in the paper. 
    Do you have any script to generate training dataset?
3. Why do you use 128x128 dimension size as training dataset instead of same dimension size as test dataset?

Thanks and best regards!

**Comment by ArthurFirmino** [comment]
Hello yyuzhongpv,

1. The dataset was a mix of publicly available scenes rendered in PBRT-v4 (from here https://github.com/mmp/pbrt-v4-scenes), and procedurally generated scenes rendered in Mitsuba. The scene generator is written in C++ and it creates a scene by randomly placing objects with random materials in front of the camera, and it outputs an XML file. If you are interested I could create a repository with the source code to the scene generator.

2. I used 128x128 images for the training dataset, so that I could support a larger number of scenes without having to render references for full resolution images which takes a long time. I don't think this negatively impacts training as the CNNs used tile over a smaller pixel area. Also, during training if you use full size images you may run into GPU memory limitations, and then you'd want to sample random smaller crops at each iteration anyways. Using larger images during training might also have an effect similar to increasing the batch size, as then each gradient descent step is the average of the gradient over a larger number of pixels. 

Note, this is mostly based on my intuition and from details I've read in other denoising papers. I did not do any systematic experimenting with different image sizes.

**Comment by yyuzhongpv** [comment]
Hi Arthur,

Thanks for your quick response with so much details.

I will highly appreciate if you can share a repository to generate the dataset for training. I spent more than a week to search the dataset that I can use to train oidn model, and saw some other guys also asked the same questions in Intel oidn repository. 

I am working on customizing the oidn model to make it lighter to run in real-time on the smart phone. Do you have any suggestions for this direction? 

Best regards.

**Comment by yyuzhongpv** [comment]
Hi Arthur,

Excuse me for another question about the dataset. 

I found some images in training folder have large displayWindow (1024x1024), but the valid data area is only small portion with 128x128. Most of other images have same displayWindow and dataWindow (128x128). Did you generate those pattern with larger displayWindow on purpose and why?  

Example with larger displayWindow:
![image](https://user-images.githubusercontent.com/30056223/192571054-207e6318-e243-47a1-badd-f71d947b7276.png)

Thanks and best regards.

**Comment by yyuzhong** [comment]
Hi Arthur,

Would you please share the codes of scene generator and dataset generator with Mitsuba as well?

Thank you so much.

**Comment by ArthurFirmino** [comment]
Sorry for the delay in replying. I have uploaded a version of the scene generator here: https://github.com/ArthurFirmino/scene-gen

To generate 100 128x128 images at 32spp you could use a simple command line script like this:
```
seq 100 | xargs -I {} sh -c "./SceneGen --seed {} --spp 32 --crop-window-size 128 > {}.xml"
seq 100 | xargs -I {} mitsuba {}.xml
```

About the 128x128 window, I believe these were PBRTv4 renders using the `--cropwindow` option which inserts some metadata into the .exr file about the cropped window. The result is that image viewers will display the image like that. It was a bit of annoyance that I only noticed after I had rendered many 128x128 crops, and then had to work around it.

I don't have many suggestions about implementing a UNet-based denoiser like OIDN on a smartphone, other than to make use of the smartphone's GPU and applicable tensor operations if available. I don't have any knowledge about ML frameworks on smartphones.

**Comment by yyuzhongpv** [comment]
Thank you so much Arthur!

For the denoise model on the smart phone, the only thing to do is to customize a lighter ML model with less computation.
There are already some ML frameworks on smart phone like TensorFlow lite, PyTorch mobile etc., which can optimize the inference time with GPU (OpenGL/OpenCL) or DSP/NPU.

Really appreciate your valuable information.
